docs/book/src/development/development.md

```diff
@@ -50,9 +50,80 @@ make docker-build docker-push
 
 After generating `infrastructure-components.yaml`, replace the `us.gcr.io/k8s-artifacts-prod/capi-openstack/capi-openstack-controller:v0.3.4` with your image.
 
+## Test Cluster Creation with the dev-test ClusterClass
+
+This guide demonstrates how to create a Kubernetes cluster using ClusterClass, specifically designed for a development environment. It includes setting up necessary environment variables, initializing a Kind cluster, configuring secrets, and applying ClusterClass to deploy your cluster.
+
+### Environment Variables Setup
+
+Configure the following environment variables for your Kubernetes cluster in OpenStack. Default values are aligned with the devstack setup (which is documented below). When setting the `KUBERNETES_VERSION`, ensure an image named ubuntu-2204-kube-{{ KUBERNETES_VERSION }} is available in your setup, matching that Kubernetes version.
+
+```bash
+# Define the name of your cluster
+export CLUSTER_NAME="<cluster_name>"
+# Desired Kubernetes Version for Your Cluster (e.g., "v1.28.5")
+export KUBERNETES_VERSION="<kubernetes_version>"
+
+# [Optional] SSH Keypair Name for Instances in OpenStack (Default: "")
+export OPENSTACK_SSH_KEY_NAME="<openstack_keypair_name>"
+# [Optional] Control Plane Machine Flavor (Default: m1.medium)
+export OPENSTACK_CONTROL_PLANE_MACHINE_FLAVOR="<openstack_control_plane_machine_flavor>"
+# [Optional] Node Machine Flavor (Default: m1.small)
+export OPENSTACK_NODE_MACHINE_FLAVOR="<openstack_node_machine_flavor>"
+# [Optional] OpenStack Cloud Environment (Default: capo-e2e)
+export OPENSTACK_CLOUD="<openstack_cloud>"
+```
+
+### Creating a Kind Cluster
+
+Create a Kind cluster. This process involves the initialization of Cluster API providers for OpenStack.
+
+```bash
+kind create cluster
+export CLUSTER_TOPOLOGY=true
+clusterctl init --infrastructure openstack
+```
+
+### Secret Configuration
+
+CAPO needs a clouds.yaml file in order to manage the OpenStack resources needed for the Cluster. This should be supplied as a secret named `${CLUSTER_NAME}-clouds-yaml`. You can create this secret for example with:
+
+```bash
+kubectl create secret generic ${CLUSTER_NAME}-clouds-yaml --from-file=clouds.yaml
```

Comment:
note that clouds.yaml secret is named `${CLUSTER_NAME}-cloud-config` in our templates, we might want to keep it consistent.

Reply:
That is right, but it is a bit confusing because we use both clouds.yaml and cloud.conf files. I had it like the templates before but @lentzi90 suggested this change. What do you say Lentzi?

Reply:
I suggest we try to keep the name consistent across the board even if it's just a name, so my preference goes over using `${CLUSTER_NAME}-cloud-config` everywhere.

Reply:
Ah I had missed that the other templates used this format. I agree it would be nice with `${CLUSTER_NAME}-cloud-config` to keep things consistent. However, I now realized an issue with this...

We cannot use the `CLUSTER_NAME` variable in the ClusterClass! It is supposed to be used to create multiple clusters with unique names. Actually now that I tested it in tilt, I get an error that the name is invalid for all the objects that has a name starting with `CLUSTER_NAME`. The names end up like `-control-plane-template`, which errors because it starts with `-`.

I think we should put a fixed name on the secret and all the other objects used by the ClusterClass. For example `dev-test`. So the secret would be named `dev-test-clouds-yaml` or `dev-test-cloud-config`. I'm fine with either.

Reply:
The name of the secret is now changed to follow the earlier templates.
Regarding the `CLUSTER_NAME`, it is only to add it into a variable in the `tilt-settings.yaml/json` file and then it should work fine. I have updated the docs about it.